libstorage/BinLogHandler.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * @CopyRight:
+ * FISCO-BCOS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2018 fisco-dev contributors.
+ */
+/** @file BinLogHandler.h
+ *  @author chaychen
+ *  @date 20190802
+ */
+
+#include "BinLogHandler.h"
+#include <json/json.h>
+#include <libdevcore/FixedHash.h>
+#include <libdevcore/easylog.h>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/crc.hpp>
+#include <boost/filesystem.hpp>
+#include <fstream>
+
+namespace fs = boost::filesystem;
+
+using namespace dev;
+using namespace dev::storage;
+
+BinLogHandler::BinLogHandler() {}
+
+BinLogHandler::~BinLogHandler()
+{
+    if (m_outBinaryFile.is_open())
+    {
+        m_outBinaryFile.close();
+    }
+}
+
+void BinLogHandler::setBinLogStoragePath(const std::string& path)
```

Comment:
why binlog need path? what about use a fixed path?

Reply:
Done. Directory splicing is implemented externally.